File: static/Leidenfrost_Intersection.py
import enum

class State(enum.Enum):
    Stable = "Stable"
    MetaStable = "MetaStable"
    CognitiveOverflow = "CognitiveOverflow"

class Agent:
    def __init__(self, reflective: bool, capacity: float, realityTesting: bool):
        self.reflective = reflective
        self.capacity = capacity
        self.realityTesting = realityTesting

class Algorithm:
    def __init__(self, engagementWeight: float, retentionWeight: float, predictivityWeight: float, demand: float):
        self.engagementWeight = engagementWeight
        self.retentionWeight = retentionWeight
        self.predictivityWeight = predictivityWeight
        self.demand = demand

def optimize(a: Algorithm, e: float, r: float, p: float) -> float:
    return a.engagementWeight * e + a.retentionWeight * r + a.predictivityWeight * p

def cognitive_load(a: Algorithm, h: Agent) -> float:
    return a.demand - h.capacity

def evaluate_state(a: Algorithm, h: Agent, t: float) -> State:
    load = cognitive_load(a, h)
    threshold = 1.0
    if not h.reflective:
        return State.Stable
    if not h.realityTesting:
        return State.CognitiveOverflow
    if load <= 0:
        return State.Stable
    if load > 0 and t < threshold:
        return State.MetaStable
    return State.CognitiveOverflow

def system_judgement(a: Algorithm, h: Agent) -> str:
    if h.reflective and a.demand > h.capacity:
        return "DESIGN_DEFICIT: algoritmo unsafe para agentes reflexivos"
    return "SYSTEM_OK: comportamento esperado"

# Exemplo concreto
agent_lucido = Agent(reflective=True, capacity=5.0, realityTesting=True)
algoritmo_plataforma = Algorithm(engagementWeight=1.0, retentionWeight=1.0, predictivityWeight=1.0, demand=7.5)

t_exposicao = 0.8
print(evaluate_state(algoritmo_plataforma, agent_lucido, t_exposicao).value)
print(system_judgement(algoritmo_plataforma, agent_lucido))

File: static/Leidenfrost_Intersection.hs
-- ============================================
-- Sistema Sociotécnico Algorítmico (SSA)
-- Modelo formal da interseção Leidenfrost
-- ============================================

-- Tipos fundamentais --------------------------

type Engagement   = Double
type Retention    = Double
type Predictivity = Double

type CognitiveCapacity = Double
type ProcessingDemand  = Double
type TimeExposure      = Double

-- Agente humano --------------------------------

data Agent = Agent
  { reflective      :: Bool          -- capacidade de metamodelagem (H*)
  , capacity        :: CognitiveCapacity
  , realityTesting  :: Bool           -- ainda valida o real
  } deriving (Show)

-- Algoritmo ------------------------------------

data Algorithm = Algorithm
  { engagementWeight   :: Double
  , retentionWeight    :: Double
  , predictivityWeight :: Double
  , demand             :: ProcessingDemand
  } deriving (Show)

-- Estado emergente ------------------------------

data State
  = Stable
  | MetaStable        -- "fio de Leidenfrost"
  | CognitiveOverflow -- estresse estrutural
  deriving (Show)

-- Função objetivo do sistema -------------------

optimize :: Algorithm -> Engagement -> Retention -> Predictivity -> Double
optimize a e r p =
    engagementWeight a   * e +
    retentionWeight a    * r +
    predictivityWeight a * p

-- Carga cognitiva ------------------------------

cognitiveLoad :: Algorithm -> Agent -> Double
cognitiveLoad a h =
    demand a - capacity h

-- Avaliação da interseção ----------------------

evaluateState :: Algorithm -> Agent -> TimeExposure -> State
evaluateState a h t
  | not (reflective h) = Stable
  | not (realityTesting h) = CognitiveOverflow
  | load <= 0 = Stable
  | load > 0 && t < threshold = MetaStable
  | otherwise = CognitiveOverflow
  where
    load = cognitiveLoad a h
    threshold = 1.0   -- tempo máximo tolerável sem isolamento

-- Juízo técnico --------------------------------

systemJudgement :: Algorithm -> Agent -> String
systemJudgement a h
  | reflective h && demand a > capacity h =
      "DESIGN_DEFICIT: algoritmo unsafe para agentes reflexivos"
  | otherwise =
      "SYSTEM_OK: comportamento esperado"

-- Exemplo concreto -----------------------------

agentLucido :: Agent
agentLucido = Agent
  { reflective = True
  , capacity = 5.0
  , realityTesting = True
  }

algoritmoPlataforma :: Algorithm
algoritmoPlataforma = Algorithm
  { engagementWeight = 1.0
  , retentionWeight = 1.0
  , predictivityWeight = 1.0
  , demand = 7.5
  }

-- Simulação ------------------------------------

main :: IO ()
main = do
  let t = 0.8 -- tempo de exposição
  print $ evaluateState algoritmoPlataforma agentLucido t
  putStrLn $ systemJudgement algoritmoPlataforma agentLucido
